misc/dbt-materialize/dbt/adapters/materialize/impl.py

```diff
@@ -178,3 +179,58 @@ def render_raw_columns_constraints(
                 rendered_column_constraints.append(" ".join(rendered_column_constraint))
 
         return rendered_column_constraints
+
+    def get_column_schema_from_query(self, sql: str) -> List[PostgresColumn]:
```

Comment:
Using temporary views is **really** smart! I'll move this out of `impl.py` and test this approach.

Reply:
Wait, move it out of impl.py? But the method is part of the `MaterializeAdapter` class.

Reply:
Nervermind; in my brain the query we wanted to override was baked into the macro.